scripts/tests/chiptest/runner.py

```diff
@@ -132,7 +132,9 @@ def RunSubprocess(self, cmd, name, wait=True, dependencies=[]):
 
         for dependency in dependencies:
             for accessory in dependency.accessories:
-                wait.add_process(accessory, dependency)
+                # Only wait on processes that have been started
+                if accessory.started == True:
```

Comment:
@vivien-apple I'm not sure if this is the best way to do this but since not all applications added to `AppsRegister` is started, waiting on that process doesn't make sense. I thought about using the `accessory.stopped` value but that doesn't quite track what I need (that an application has been started). Thoughts on this?

Reply:
I would prefer to hide that into the `wait` method of accessory. So this code will stay independent of that.

About using the stopped flag I'm not sure why the accessory is inited with `self.stopped = False`. If we change that to `True` by default does it helps ?

Reply:
> About using the stopped flag I'm not sure why the accessory is inited with self.stopped = False. If we change that to True by default does it helps ?
Thanks, I thought that was intentional so didn't mess with it. Making it `True` by default does help. I'll use `self.stopped` instead of introducing a new value.

Reply:
> I would prefer to hide that into the wait method of accessory. So this code will stay independent of that.

So the issue is I can't NOT wait on the any applications that have been added to the apps_register. In runner.py, if our application is in the apps_register but not running, then it gets added to the queue here: https://github.com/project-chip/connectedhomeip/blob/master/scripts/tests/chiptest/runner.py#L96

This then causes this Exception to occur every time `RunSubprocess` is called: https://github.com/project-chip/connectedhomeip/blob/master/scripts/tests/chiptest/runner.py#L142. This includes things like doing that initial pairing for the default application.

My new solution is to just start all the application that have been added to the apps_register (making sure chip-tool is not added). Let me know what you think.

Reply:
Thanks for the suggestion. Adding the check in `wait` to sleep when process does not exist is better than starting all apps.